src/app/comic/create/publish/BranchPlacer.tsx

```diff
@@ -0,0 +1,140 @@
+"use client";
+
+import {useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import PublishButton from "./PublishButton";
+import {PanelSet, Branch} from "./PanelSet";
+
+
+const BranchPlacer = () => {
+    const [branCount, setBranch] = useState(0);
+    const [adding, setAdding] = useState(false);
+    const [parId, setparId] = useState(" ");
+    const [curId, setCurId] = useState(parId ? parId : uuidv4());
+
+    const baseBranch = {
+        panel: -1,
+        x: 0,
+        y: 0,
+        child_branch_uuid: "string"
+    }
+
+    const startPs = {
+        current_panel_set_uuid: curId,
+        parent_branch_uuid: parId,
+        image_paths: ["","",""],
+        branches:[baseBranch,baseBranch,baseBranch]
+    }
+
+    const [ps, setPs] = useState<PanelSet>(startPs);
+    
+    const startAdd = () => {
+        console.log("You have reached maximum amount of branches");
+        if(branCount < 3 ){
+            console.log("Start adding");
+            setAdding(true);
+        }
+    }
+
+    const addHook = (panel:number) => {
+        if(!adding){ //makes sure the user clicked add hook before 
+            console.log("you aren't adding yet");
+            return;
+        }
+        console.log("Adding to panel " + panel);
+        let bholder = ps.branches; //temporary holding variable for the branches array for editing
+        for(let i = 0; i < 3; i++){
+            if(i == branCount){
+                bholder[i].panel = panel; //adds that branch was added from panel 1
+                bholder[i].x = 50 + 100 * branCount; //x-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].y = 100; //y-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].child_branch_uuid = uuidv4();
+            }
+            console.log(i);
+        }
+        
+        setPs((prevPs) => {
+            return {
+                current_panel_set_uuid: prevPs.current_panel_set_uuid,
+                parent_branch_uuid: prevPs.parent_branch_uuid,
+                image_paths: prevPs.image_paths,
+                branches: bholder //updating branches with new hook
+            }
+        });
+        
+        setBranch(branCount + 1);
+        console.log("branch hook added. Total branch hooks: " + (branCount + 1));
+        setAdding(false);
+        console.log("No longer adding");
+    }
+
+    const removeHook = () => {
+        let bholder = ps.branches;
+        for(let i = 0; i < 3; i++){
+            if(i == branCount){
+                bholder[i].panel = -1; //since not on a panel set's hook to -1
+                bholder[i].x = 0; //x-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].y = 0; //y-value of hook -----> TODO: align hook with mouse placement on panel
+                bholder[i].child_branch_uuid = uuidv4();
+                break;
```

Comment:
May need to call setPs again to persist these changes

Reply:
yes good catch i just fixed this